bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/events/BahmniEvent.java

```diff
@@ -0,0 +1,35 @@
+package org.bahmni.module.bahmnicore.events;
+
+import org.openmrs.api.context.UserContext;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+
+@Component
```

Comment:
Should not be spring component. You can create a constructor with the eventType and encounter object and set internal attributes